File: src/Helpers/validator.php
<?php

function validate_input($input) {
    if (empty($input)) {
        throw new InvalidArgumentException('Invalid');
    }
    return true;
}

function sanitize_input($input) {
    return htmlspecialchars($input);
}

function check_required($data, $fields) {
    foreach ($fields as $field) {
        if (!isset($data[$field])) return false;
    }
    return true;
}

function check_length($input, $max = 255) {
    return strlen($input) <= $max;
}

File: src/main/kotlin/helpers/Validator.kt
package helpers

fun validateInput(input: String): Boolean {
    if (input.isEmpty()) throw IllegalArgumentException("Invalid")
    return true
}

fun sanitizeInput(input: String): String {
    return input.replace("<", "").replace(">", "")
}

fun checkLength(input: String, max: Int = 255): Boolean {
    return input.length <= max
}

fun normalizeInput(input: String): String {
    return input.trim().lowercase()
}
